backend/src/api/model/realm/mutations.rs

```diff
@@ -24,11 +24,18 @@ impl Realm {
             return Err(invalid_input!(key = "realm.path-is-reserved", "path is reserved and cannot be used"));
         }
 
+        let AuthContext::User(user) = &context.auth else { unreachable!() };
+        let roles: Vec<&String> = if !parent.is_current_user_owner(context) {
+            vec![&user.user_role]
+        } else {
+            vec![]
+        };
```

Comment:
Is the type declaration really necessary? Also, the `unreachable!` line has to be inside the `if`, otherwise we haven't really achieved much. Only then do we know that we have a user.